examples/hosts/app-integration/external-data/src/model/taskList.ts

```diff
@@ -10,79 +10,77 @@ import { IFluidHandle } from "@fluidframework/core-interfaces";
 import { SharedString } from "@fluidframework/sequence";
 import { SharedMap } from "@fluidframework/map";
 
-import type { ITask, ITaskEvents, ITaskList, TaskData } from "../model-interface";
+import type {
+	ExternalSnapshotTask,
+	ITask,
+	ITaskEvents,
+	ITaskList,
+	TaskData,
+} from "../model-interface";
 import { externalDataServicePort } from "../mock-external-data-service-interface";
 
 class Task extends TypedEventEmitter<ITaskEvents> implements ITask {
 	public get id(): string {
 		return this._id;
 	}
 	// Probably would be nice to not hand out the SharedString, but the CollaborativeInput expects it.
-	public get name(): SharedString {
-		return this._name;
+	public get draftName(): SharedString {
+		return this._draftName;
 	}
-	public get priority(): number {
-		const cellValue = this._priority.get();
+	public get draftPriority(): number {
+		const cellValue = this._draftPriority.get();
 		if (cellValue === undefined) {
 			throw new Error("Expected a valid priority");
 		}
 		return cellValue;
 	}
-	public get incomingName(): string | undefined {
-		return this._incomingName;
+	public set draftPriority(newPriority: number) {
+		this._draftPriority.set(newPriority);
 	}
-	public set incomingName(newValue: string | undefined) {
-		this._incomingName = newValue;
-		this.emit("incomingNameChanged");
+	public get externalDataSnapshot(): ExternalSnapshotTask {
+		return this._externalDataSnapshot;
 	}
-	public get incomingType(): string | undefined {
-		return this._incomingType;
-	}
-	public set incomingType(newValue: string | undefined) {
-		this._incomingType = newValue;
-	}
-	public get incomingPriority(): number | undefined {
-		return this._incomingPriority;
-	}
-	public set incomingPriority(newValue: number | undefined) {
-		this._incomingPriority = newValue;
-		this.emit("incomingPriorityChanged");
+	public set externalDataSnapshot(newValue: ExternalSnapshotTask) {
+		const changesAvailable = newValue.changeType !== undefined;
+		this._externalDataSnapshot.changeType = newValue.changeType;
+		this._externalDataSnapshot.id = newValue.id;
+		if (this._externalDataSnapshot.name !== newValue.name) {
+			this._externalDataSnapshot.name = newValue.name;
+		}
+		if (this._externalDataSnapshot.priority !== newValue.priority) {
+			this._externalDataSnapshot.priority = newValue.priority;
+		}
```

Comment:
Feels like we can just blindly overwrite all properties. No special events or anything emitted by changes to name or priority that we might want to avoid, right? And if so, can we just do `this._externalDataSnapshot = { ... newValue}`?